admin/admin-functions.php

```diff
@@ -0,0 +1,139 @@
+<?php
+/**
+ * The admin-specific functionality of the plugin.
+ *
+ * @link       https://nataliesmith.dev
+ * @since      1.0.0
+ * @package    Poetwp
+ * @subpackage Poetwp/admin
+ */
+
+/**
+ * Add custom meta boxes to the poem post type
+ */
+function poetwp_add_meta_boxes() {
+	add_meta_box( 'poem_details', 'Poem Details', 'poetwp_poem_details_callback', 'poem', 'normal', 'default' );
+}
+// Hook the meta box function to the 'add_meta_boxes' action.
+add_action( 'add_meta_boxes', 'poetwp_add_meta_boxes' );
+
+/**
+ * Callback function to display the contents of the meta box
+ *
+ * @param WP_Post $post The current post object.
+ */
+function poetwp_poem_details_callback( $post ) {
+	// Add a nonce field for security.
+	wp_nonce_field( 'poetwp_save_poem_details', 'poetwp_poem_details_nonce' );
+
+	// Retrieve existing meta data.
+	$date  = get_post_meta( $post->ID, 'poem_date', true );
+	$notes = get_post_meta( $post->ID, 'poem_notes', true );
+
+	// Output the form fields.
+	echo '<label for="poem_date">Date: </label>';
+	echo '<input type="text" id="poem_date" name="poem_date" class="poetwp-datepicker" value="' . esc_attr( $date ) . '" size="25" />';
+	echo '<label for="poem_notes">Notes: </label>';
```

Comment:
Rather than using PHP to echo HTML, it's generally preferrable to print HTML and insert the PHP bits where needed:

```suggestion
	?>
	<label for="poem_date"><?php esc_html_e( 'Date:', 'poetwp' ); ?></label>
	<input type="text" id="poem_date" name="poem_date" class="poetwp-datepicker" value="<?php echo esc_attr( $date ); ?>" size="25" />
	<label for="poem_notes"><?php esc_html_e( 'Notes:', 'poetwp' ); ?></ladel>
	<?php
```

Why? 

Because long strings in PHP are tricky to debug; having the IDE syntax highlight parts of the HTML it can identify as HTML is much nicer - you might not have spotted the intentional `ladel` in that example above, but it definitely would have stood out in your IDE.

Reply:
Noted #24